krisHutchison/answers.md

```diff
@@ -0,0 +1,240 @@
+
+
+**A:** 
+`(t-b) > (r-l) ? (r-l) * 3.14 : (t-b) * 3.14`
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+
+**A:** 
+`(Math.sqrt((Math.pow((t-b), 2) + Math.pow((r-l), 2))) *3.14);`
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+
+**A:** 
+
+	leftHorizontal = ((r-l) / 3 ) - l  
+	rightHorizontal = ((r-l) /3 ) - r
+	bottomVertical = ((t-b) / 3 ) -b
+	topVertical = ((t-b) / 3 ) -t 
+
+---
+
+**4)** (_Difficulty: easyish_)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+* Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+* Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+**a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C.
+
+
+**A:** 
+`N = (R * 8) + C`
+
+**b)**  Given N, find R.  Write an expression for R which contains N.
+
+
+**A:** 
+`R = N / 8`
+
+**c)**  Given N, find C.  Write an expression for C which contains N.
+
+
+**A:** 
+`C = N % 8`
+
+**d)**  Assume the squares are colored black and white, with the upper-left square black.
+Write an expression to set a variable _color_ to either 'black' or 'white', describing the square identified by variables R,C, and N.
+(_Hint: If you prefer, you may use a pair of conditional statements instead of a single expression._)
+
+
+**A:** 
+`((n % 2 == 0) && (r % 2 == 0)) ? n="black" : n="white"`
+
+---
+
+**5)** (_Difficulty: moderate_)
+
+Suppose you represent a fraction (_n/d_) with 2 integer variables: _n_ for the numerator and _d_ for the denominator.
+If _n_ is greater than _d_, the fraction is "improper", but it can be rewritten as a proper fraction.  For example, "7/4" is improper, but it can be rewritten as "1 3/4", which is proper.
+
+Write an expression for a string expressing the proper form of an improper fraction _n/d_.  For example, when _n==7_ and _d==4_, your resulting string should be "1 3/4".  You may assume both _n_ and _d_ are positive integers and _n_ > _d_, but otherwise you should be able to handle any values of _n_ and _d_.
+(_Hint: you'll need the modulo operator _%_, and you'll probably want to create a couple of extra variables._)
+
+
+**A:** 
+
+	whole = Math.floor(n/d); 
+	rem = Math.round( n % d ); 
+	
+	n > d ? whole + " " + rem + "/" + d : n+"/"+d
```

Comment:
Nice work! Only thing I'd say is that you don't really need to round `rem`, as % should just return whole numbers anyway.